language/go/package.go

```diff
@@ -166,6 +169,54 @@ func (pkg *goPackage) inferImportPath(c *config.Config) error {
 	return nil
 }
 
+// Returns a a suitable go_library name based on the dir.
+// Major version suffixes (eg. "v1") are dropped.
+func libNameFromDir(dir string) string {
```

Comment:
Change name and signature to `libNameFromImportPath(imp string) (string, bool)`. If we can't find the library name because the import path is empty or garbled, let's return false. Callers may want to handle that.

Maybe the regexp can just match the last component, dropping the version suffix? Haven't tested this, but:

```
^(?:.*/)([^/]+)(?:/v\d+)?$
```

That could simplify the function to:

```
m := goPackageRe.FindStringSubmatch(imp)
if m == nil {
  return "", false
}
return m[1], true
```